stock_package_nbr/models/stock_picking.py

```diff
@@ -0,0 +1,24 @@
+# Copyright 2017 Sylvain Van Hoof (Okia) <sylvain@okia.be>
+# Copyright 2018-2019 Jacques-Etienne Baudoux (BCIM) <je@bcim.be>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import fields, models
+
+
+class StockPicking(models.Model):
+    _inherit = 'stock.picking'
+
+    def put_in_pack(self):
+        self.ensure_one()
+        if not self.move_line_ids.filtered(
+                lambda o: o.qty_done > 0 and not o.result_package_id):
+            # let's raise the standard error
+            return super(StockPicking, self).put_in_pack()
+
+        res = self.check_destinations()
+        if res.get('type'):
+            return res
+
+        return self.env.ref('stock_package_nbr.put_in_pack_nbr').read()[0]
```

Comment:
I think this behaviour should be configurable. IMHO, an option on stock.picking.type level is a good way of doing this.

Reply:
I completely agree but I leave it as an improvement